compat/mingw.c

```diff
@@ -810,13 +810,24 @@ static int mingw_open_existing(const wchar_t *filename, int oflags, ...)
 			     &security_attributes, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
 	if (handle == INVALID_HANDLE_VALUE) {
 		DWORD err = GetLastError();
+		if (err == ERROR_ACCESS_DENIED) {
+			DWORD attrs = GetFileAttributesW(filename);
```

Comment:
We could probably also check `GetFileAttributesW()` before the first `CreateFileW()` and avoid calling `CreateFileW()` twice.

Reply:
I think the current version is fine because it optimizes for the common case, avoiding the extra `GetFileAttributesW()` & `CreateFileW()` calls for the vast majority of calls.